src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.cs

```diff
@@ -134,21 +137,28 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
             }
         }
 
+        private static (string text, string suffix) SplitMemberName(string memberString)
+        {
+            for (var i = 0; i < memberString.Length; i++)
+            {
+                if (!SyntaxFacts.IsIdentifierPartCharacter(memberString[i]))
+                    return (memberString[0..i], memberString[i..]);
+            }
+
+            return (memberString, "");
+        }
+
         protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
             => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
 
         public override Task<TextChange?> GetTextChangeAsync(
             Document document, CompletionItem selectedItem, char? ch, CancellationToken cancellationToken)
         {
-            if (ch == '(')
-            {
-                if (selectedItem.Properties.TryGetValue(InsertionTextOnOpenParen, out var insertionText))
-                {
-                    return Task.FromResult<TextChange?>(new TextChange(selectedItem.Span, insertionText));
-                }
-            }
+            var insertionText = ch == '(' && selectedItem.Properties.TryGetValue(InsertionTextOnOpenParen, out var value)
```

Comment:
Looks like there was an existing bug here: this should also consider [ if you're typing an indexer, right? It seems we have different behavior.

Reply:
sure.  i can update taht.